### PR TITLE
Fix login

### PR DESCRIPTION
**Pull request by MohamedElmdary**

### CODE REVIEW

File: client/src/clients/api/index.ts [hunk 1]
```diff
@@ -47,6 +47,15 @@ export class ApiClient extends ApiClientBase {
     this.requestInterceptorId = this.setAxiosRequestInterceptor()
     ApiClientBase.$api = this
 
+    try {
+      const user = JSON.parse(atob(localStorage.getItem(ApiClientBase.USER_KEY) || '') || '""')
```
Comment (zaelgohary) [inline client/src/clients/api/index.ts:51]:
```suggestion
      const user = JSON.parse(atob(localStorage.getItem(ApiClientBase.USER_KEY) || '') || '')
```

File: client/src/clients/api/base.ts [hunk 2]
```diff
@@ -28,34 +26,28 @@ export abstract class ApiClientBase {
...
     ApiClientBase.assertUser()
-    ApiClientBase.USER = null
+    localStorage.removeItem(ApiClientBase.USER_KEY)
+    ApiClientBase.USER.value = null
   }
 
   protected static assertUser() {
-    const token = localStorage.getItem("access_token")
-    if (!ApiClientBase.USER && !token) {
+    if (!ApiClientBase.USER) {
```
Comment (zaelgohary) [inline client/src/clients/api/base.ts]:
```suggestion
    if (!ApiClientBase.USER.value) {
```

File: client/src/clients/api/index.ts (hunk 1 above)
Comment (MohamedElmdary) [reply client/src/clients/api/index.ts:51]: `''` is not a valid input for JSON.parse so it will throw

Comment (MohamedElmdary) [reply client/src/clients/api/index.ts:51]: image

**Review by zaelgohary — APPROVED** [review]

**Pull request merged**